devito/dle/backends/advanced.py

```diff
@@ -158,9 +158,9 @@ def _loop_blocking(self, nodes, state):
                 dim = blocked.setdefault(i, Dimension(name=name))
                 block_size = dim.symbolic_size
                 iter_size = i.dim.symbolic_extent
-                start = i.limits[0] - i.offsets[0]
-                finish = i.dim.symbolic_end - i.offsets[1]
-                innersize = iter_size - (-i.offsets[0] + i.offsets[1])
+                start = i.limits[0] + i.offsets[0]
+                finish = i.dim.symbolic_end + i.offsets[1]
+                innersize = iter_size + (-i.offsets[0] + i.offsets[1])
```

Comment:
So this PR already includes the domain switch?

Reply:
> So this PR already includes the domain switch?

no (see description and see my answer to @mloubout below); it does stuff necessary/useful for the switch, but we're still far from it

> Wait, why does + becomes - ?

This PR fixes a stupid hack regarding ``Iteration`` offsets; see https://github.com/opesci/devito/pull/469/files#diff-60ab6d9da10e52ca6ffa3e60cec1dc11L279
In essence, previously we were treating ``offsets`` as relative to the data space, rather than the iteration space. Now we pass in the appropriate information, so we don't have to negate it all over the place anymore.
Example:
previously we were passing ``offsets=(-3, 3)``; now we pass ``offsets=(3, 3)``